examples/distribute-across-az/init.pp

```diff
@@ -0,0 +1,74 @@
+# Feel free to change these variables to launch a different number
+# of instances or launch them in different availability zones
+$availability_zones = ['sa-east-1a', 'sa-east-1b']
+$number_of_app_servers = 3
+$maximum_app_servers = 10
+$region = 'sa-east-1'
+$ami = 'ami-67a60d7a'
+
+# Don't change below this line
+# First lets validate the variables passed in
+validate_array($availability_zones)
+validate_string($region)
+validate_string($ami)
```

Comment:
Since we're already using future parser, we could use assert_type:
```puppet
assert_type(Array[String[1]], $availability_zones)
assert_type(String[1], $region)
assert_type(String[1], $ami)
```

The String[1] validates that it's a non-empty string.